### PR TITLE
refactor(ui): icons integration

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

### CODE REVIEW

File: src/framework/ui/button/button.component.tsx [hunk 1]
```diff
@@ -24,7 +22,7 @@ import {
 } from './type';
 
 interface ButtonProps {
-  icon?: ImageSourcePropType;
+  icon?: (width: number, height: number, color: string) => React.ReactElement<ImageProps>;
```
Comment (artyorsh) [inline src/framework/ui/button/button.component.tsx]: why not using style in arguments?

File: src/framework/ui/button/button.component.tsx [hunk 2]
```diff
@@ -93,15 +91,10 @@ export class Button extends React.Component<Props> {
...
-
-    return (
-      <Image
-        key={0}
-        style={[style, strictStyles.icon]}
-        source={image}
-      />
-    );
+    const { icon } = this.props;
+    return icon && React.cloneElement(icon(style.width, style.height, style.color), {
```
Comment (artyorsh) [inline src/framework/ui/button/button.component.tsx]: * prefer using control flow operators like `if`
* assign icon value to variable like `iconElement`

File: src/framework/ui/button/type.ts [hunk 3]
```diff
@@ -43,6 +43,9 @@ export class ButtonAlignments {
   }
 
   private static typeOf(value: any): value is ButtonAlignment {
+    if (!value) {
```
Comment (artyorsh) [inline src/framework/ui/button/type.ts]: Fixed with `ButtonGroup` implementation. Can be omitted

File: src/framework/ui/tab/tab.component.tsx [hunk 4]
```diff
@@ -41,38 +39,32 @@ export class Tab extends React.Component<Props> {
...
-      icon: {},
+      icon: source.icon,
       title: source.text,
     };
   };
 
-  private createTextComponent = (style: StyleType): React.ReactElement<TextProps> => {
-    return (
+  private createTextComponent = (style: StyleType): React.ReactElement<TextProps> | null => {
+    return this.props.title ? (
```
Comment (artyorsh) [inline src/framework/ui/tab/tab.component.tsx]: const hasTitle: boolean = this.props.title !== undefined;

return hasTitle ? this.renderTitle() : undefined;

File: src/framework/ui/tab/tab.component.tsx [hunk 5]
```diff
@@ -41,38 +39,32 @@ export class Tab extends React.Component<Props> {
...
-    const { icon, title } = this.props;
-
-    return [
-      icon ? this.createImageComponent(style.icon) : undefined,
-      title ? this.createTextComponent(style.title) : undefined,
-    ];
+  private createImageComponent = (style: StyleType): React.ReactElement<ImageProps> | null => {
+    const icon: React.ReactElement<ImageProps> =
+      this.props.icon && this.props.icon(style.width, style.height, style.color);
+    return icon ? React.cloneElement(icon, { key: 2 }) : null;
```
Comment (artyorsh) [inline src/framework/ui/tab/tab.component.tsx]: same code style as for title

File: src/framework/ui/tab/tab.component.tsx [hunk 6]
```diff
@@ -41,38 +39,32 @@ export class Tab extends React.Component<Props> {
...
   };
 
+  private createComponentChildren = (style: StyleType): React.ReactNode => ([
+    this.createImageComponent(style.icon),
```
Comment (artyorsh) [inline src/framework/ui/tab/tab.component.tsx]: You can check for children creation need right here.
See [example](https://github.com/akveo/react-native-ui-kitten/blob/next/src/framework/ui/tab/tab.component.tsx#L67)

File: src/playground/src/ui/screen/button.component.tsx [hunk 7]
```diff
@@ -45,39 +46,79 @@ class Button extends React.Component<Props, State> {
               status={STATUS}
               size='giant'
               alignment={ALIGNMENT}
-              icon={ICON}
+              icon={(width: number, height: number, color: string) => (
```
Comment (artyorsh) [inline src/playground/src/ui/screen/button.component.tsx]: refactor to component `renderIcon` variable

File: src/framework/ui/button/button.component.tsx (hunk 1 above)
Comment (32penkin) [reply src/framework/ui/button/button.component.tsx]: Done.

File: src/framework/ui/button/button.component.tsx (hunk 2 above)
Comment (32penkin) [reply src/framework/ui/button/button.component.tsx]: Done.

File: src/framework/ui/button/type.ts (hunk 3 above)
Comment (32penkin) [reply src/framework/ui/button/type.ts]: Done

File: src/framework/ui/tab/tab.component.tsx (hunk 4 above)
Comment (32penkin) [reply src/framework/ui/tab/tab.component.tsx]: Don't like this. Needs discussion.

File: src/framework/ui/tab/tab.component.tsx (hunk 5 above)
Comment (32penkin) [reply src/framework/ui/tab/tab.component.tsx]: The same

File: src/framework/ui/tab/tab.component.tsx (hunk 6 above)
Comment (32penkin) [reply src/framework/ui/tab/tab.component.tsx]: Don't see the difference.

File: src/playground/src/ui/screen/button.component.tsx (hunk 7 above)
Comment (32penkin) [reply src/playground/src/ui/screen/button.component.tsx]: Let's ask @malashkevich.

**Pull request merged**